### PR TITLE
GJS: Add tests to ensure flat-configs work with the parser

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Some debugging / exploration of flat configs to try to debug: https://github.com/ember-cli/eslint-plugin-ember/issues/2061

### CODE REVIEW

File: test-projects/configs/flat-js/eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      ember: e,
+    },
+    rules: {
+      ...ember.rules,
+      ...emberGJS.rules,
+    },
+  },
+  {
+    files: ['**/*.gjs'],
+    languageOptions: {
```
Comment (patricklx) [inline test-projects/configs/flat-js/eslint.config.mjs:44]: Gjs needs the same parser config like *.js

Comment (patricklx) [reply test-projects/configs/flat-js/eslint.config.mjs:44]: It's normally not a problem since a config file is used

Comment (patricklx) [reply test-projects/configs/flat-js/eslint.config.mjs:44]: O guess we should document it. That internally babel parser is used. Or should we just add some default plugins?

Comment (NullVoxPopuli) [reply test-projects/configs/flat-js/eslint.config.mjs:44]: I think we should document it, and then let eslint-plugin-ember decide if some utilities or defaults would be helpful to configure

**Pull request merged**